tests/integration/components/polaris-form-test.js

```diff
@@ -0,0 +1,175 @@
+      target=target
+    }}`);
+
+    assert.dom('[data-test-form]').exists();
+    assert
+      .dom('[data-test-form]')
+      .hasAttribute(
+        'accept-charset',
+        acceptCharset,
+        'renders acceptCharset attribute when provided'
+      );
+    assert
+      .dom('[data-test-form]')
+      .hasAttribute('action', action, 'renders action attribute when provided');
+    assert
+      .dom('[data-test-form]')
+      .hasAttribute(
+        'autoComplete',
+        'off',
+        'renders autoComplete attribute when provided'
+      );
+    assert
+      .dom('[data-test-form]')
+      .hasAttribute(
+        'encType',
+        encType,
+        'renders encType attribute when provided'
+      );
+    assert
+      .dom('[data-test-form-button]')
+      .exists('renders implicit submit button');
+    assert
+      .dom('[data-test-form]')
+      .hasAttribute('method', method, 'renders method attribute when provided');
+    assert
+      .dom('[data-test-form]')
+      .hasAttribute('name', name, 'renders name attribute when provided');
+    assert
+      .dom('[data-test-form]')
+      .hasAttribute(
+        'novalidate',
+        '',
+        'renders novalidate attribute when provided'
+      );
+    assert
+      .dom('[data-test-form]')
+      .hasAttribute('target', target, 'renders target attribute when provided');
+
+    this.set('implicitSubmit', false);
+
+    assert
+      .dom('[data-test-form-button]')
+      .doesNotExist(
+        'does not render submit button when `implicitSubmit` is false'
+      );
+  });
+
+  test('invokes `onSubmit` when form is submitted', async function(assert) {
+    assert.expect(1);
+
+    this.onSubmitSpy = () => {
+      assert.ok('invokes `onSubmit` action');
+    };
+
+    await render(hbs`{{polaris-form
+      onSubmit=(action onSubmitSpy)
+    }}`);
+
+    await triggerEvent('[data-test-form]', 'submit');
+  });
+
+  module('preventDefault', function() {
+    test('allows default handler to handle submit event when set to false', async function(assert) {
+      assert.expect(0);
+
+      this.onSubmitSpy = () => {
+        assert.notOk('does not invoke `onSubmit`');
+      };
+      this.preventDefaultSpy = () => {
+        assert.notOk('does not prevents default');
+      };
+
+      await render(hbs`
+        <form on-submit={{action preventDefaultSpy}}>
+          {{polaris-form
+            preventDefault=false
+            onSubmit=(action onSubmitSpy)
+          }}
+        </form>
+      `);
+
+      await triggerEvent('[data-test-form]', 'submit');
+    });
+
+    test('submits the form preventing default when not provided', async function(assert) {
+      assert.expect(0);
+
+      this.onSubmitSpy = () => {
+        assert.ok('invokes `onSubmit`');
+      };
+      this.preventDefaultSpy = () => {
+        assert.ok('prevents default `submit`');
+      };
+
+      await render(hbs`
+        <form on-submit={{action preventDefaultSpy}}>
+          {{polaris-form
```

Comment:
So tempted to use angled bracket form of this

Reply:
tbh, I've tried it out when it came out....it's pretty neat, first for easily working with tags, great auto-completion for tags as you know....and that `...attributes` curying in templates 😍

Reply:
just curious, do you know if that depends on consuming app's Ember version supporting it?

Reply:
@sivakumar-kailasam I'd want to be 100% certain that switching to angle-bracket syntax won't introduce any issues in host apps before making the jump, and then when we do so, I'd suggest we run a codemod on the entire addon to bring everything in line in one PR. Thoughts?

Reply:
Migrating in one PR sounds great. If we included [ember-angle-bracket-invocation-polyfill](https://github.com/rwjblue/ember-angle-bracket-invocation-polyfill) as a dependency of this addon then we wouldn't have to worry about breaking compatibility.